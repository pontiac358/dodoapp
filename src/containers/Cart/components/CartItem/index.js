import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled'
import CountControl from '../../../../widgets/CountControl'

const Container = styled.div`
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    align-items: center;
    margin-bottom:30px;
    @media only screen and (max-width: 480px) {
        display:block;   
    }
`

const ImageWrap = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    margin-right:20px;
    @media only screen and (max-width: 480px) {
        margin-right:0;
    }
`
const Img = styled.img`
    max-width: 150px;
    max-height: 150px;
    display:block;
    width:100%;
`

const TotalPrice = styled.div`
    font-size:25px;
    margin-left:auto;
    @media only screen and (max-width: 480px) {
        text-align:center;  
    }
`;
TotalPrice.displayName = 'TotalPrice';

const DeleteWrap = styled.div`
    font-size:25px;
    margin-left:auto;
    @media only screen and (max-width: 480px) {
        text-align:center;  
    }
`

const CountControlWrap = styled.div`
    @media only screen and (max-width: 480px) {
        text-align:center;  
    }
`;

const Title = styled.div`
    text-align: center;
    padding-right:20px;
    @media only screen and (max-width: 480px) {
       padding-right:0;

    }
`;
Title.displayName = 'Title';


const CartItem = ({ product, count, totalPrice, removeLineCart }) => {

    return (
        <Container>
            <div>
                <ImageWrap>
                    <Img src={ product.img }/>
                </ImageWrap>
                <Title>{product.title}</Title>
            </div>
            <CountControlWrap>
                <CountControl  product={product}/>
            </CountControlWrap>
            <TotalPrice>
                { totalPrice } р
            </TotalPrice>
            <DeleteWrap>
                <button onClick={()=>removeLineCart(product.id)}>Удалить</button>
            </DeleteWrap>
        </Container>
    );
}

CartItem.propTypes = {
    product: PropTypes.object,
    count: PropTypes.number
}

export default CartItem;