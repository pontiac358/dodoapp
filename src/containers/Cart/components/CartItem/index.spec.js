import React from 'react'
import { shallow } from 'enzyme'
import CartItem   from '../CartItem'

describe('CartItem component', () => {
    const props = {
        product: {
            title: 'test',
            img: '/test.png',
            price: 100
        },
        count: 1,
        totalPrice: 100
    };

    const wrapper = shallow(<CartItem {...props} />);

    it("render correctly CartItem container", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("render correctly cart title prop", () => {
        expect(wrapper.find('Title').at(0).text()).toEqual(props.product.title);
    });

    it("render correctly cart img prop", () => {
        expect(wrapper.find({ src: props.product.img })).toHaveLength(1)
    });

    it("render correctly cart price prop", () => {
        expect(wrapper.find('TotalPrice').at(0).text()).toEqual(props.totalPrice + ' Ñ€');
    });
});

