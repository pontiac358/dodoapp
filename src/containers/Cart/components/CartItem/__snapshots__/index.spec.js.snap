// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component render correctly CartItem container 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CartItem
    count={1}
    product={
      Object {
        "img": "/test.png",
        "price": 100,
        "title": "test",
      }
    }
    totalPrice={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div>
          <ForwardRef(render)>
            <ForwardRef(render)
              src="/test.png"
            />
          </ForwardRef(render)>
          <ForwardRef(render)>
            test
          </ForwardRef(render)>
        </div>,
        <ForwardRef(render)>
          <Connect(CountControl)
            product={
              Object {
                "img": "/test.png",
                "price": 100,
                "title": "test",
              }
            }
          />
        </ForwardRef(render)>,
        <ForwardRef(render)>
          100
           р
        </ForwardRef(render)>,
        <ForwardRef(render)>
          <button
            onClick={[Function]}
          >
            Удалить
          </button>
        </ForwardRef(render)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForwardRef(render)>
              <ForwardRef(render)
                src="/test.png"
              />
            </ForwardRef(render)>,
            <ForwardRef(render)>
              test
            </ForwardRef(render)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ForwardRef(render)
                src="/test.png"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "src": "/test.png",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "img",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "
    max-width: 150px;
    max-height: 150px;
    display:block;
    width:100%;
",
                ],
                "defaultProps": undefined,
                "displayName": "Styled(img)",
                "render": [Function],
                "withComponent": [Function],
              },
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "div",
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                "
    display: flex;
    align-items: center;
    justify-content: center;
    margin-right:20px;
    @media only screen and (max-width: 480px) {
        margin-right:0;
    }
",
              ],
              "defaultProps": undefined,
              "displayName": "Styled(div)",
              "render": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "test",
            },
            "ref": null,
            "rendered": "test",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "div",
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                "
    text-align: center;
    padding-right:20px;
    @media only screen and (max-width: 480px) {
       padding-right:0;

    }
",
              ],
              "defaultProps": undefined,
              "displayName": "Title",
              "render": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Connect(CountControl)
            product={
              Object {
                "img": "/test.png",
                "price": 100,
                "title": "test",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "product": Object {
              "img": "/test.png",
              "price": 100,
              "title": "test",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "
    @media only screen and (max-width: 480px) {
        text-align:center;  
    }
",
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            100,
            " р",
          ],
        },
        "ref": null,
        "rendered": Array [
          100,
          " р",
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "
    font-size:25px;
    margin-left:auto;
    @media only screen and (max-width: 480px) {
        text-align:center;  
    }
",
          ],
          "defaultProps": undefined,
          "displayName": "TotalPrice",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <button
            onClick={[Function]}
          >
            Удалить
          </button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Удалить",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Удалить",
          "type": "button",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "
    font-size:25px;
    margin-left:auto;
    @media only screen and (max-width: 480px) {
        text-align:center;  
    }
",
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "__emotion_base": "div",
      "__emotion_forwardProp": undefined,
      "__emotion_real": [Circular],
      "__emotion_styles": Array [
        "
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    align-items: center;
    margin-bottom:30px;
    @media only screen and (max-width: 480px) {
        display:block;   
    }
",
      ],
      "defaultProps": undefined,
      "displayName": "Styled(div)",
      "render": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            <ForwardRef(render)>
              <ForwardRef(render)
                src="/test.png"
              />
            </ForwardRef(render)>
            <ForwardRef(render)>
              test
            </ForwardRef(render)>
          </div>,
          <ForwardRef(render)>
            <Connect(CountControl)
              product={
                Object {
                  "img": "/test.png",
                  "price": 100,
                  "title": "test",
                }
              }
            />
          </ForwardRef(render)>,
          <ForwardRef(render)>
            100
             р
          </ForwardRef(render)>,
          <ForwardRef(render)>
            <button
              onClick={[Function]}
            >
              Удалить
            </button>
          </ForwardRef(render)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ForwardRef(render)>
                <ForwardRef(render)
                  src="/test.png"
                />
              </ForwardRef(render)>,
              <ForwardRef(render)>
                test
              </ForwardRef(render)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <ForwardRef(render)
                  src="/test.png"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "src": "/test.png",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "img",
                  "__emotion_forwardProp": undefined,
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    "
    max-width: 150px;
    max-height: 150px;
    display:block;
    width:100%;
",
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(img)",
                  "render": [Function],
                  "withComponent": [Function],
                },
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "
    display: flex;
    align-items: center;
    justify-content: center;
    margin-right:20px;
    @media only screen and (max-width: 480px) {
        margin-right:0;
    }
",
                ],
                "defaultProps": undefined,
                "displayName": "Styled(div)",
                "render": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "test",
              },
              "ref": null,
              "rendered": "test",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "
    text-align: center;
    padding-right:20px;
    @media only screen and (max-width: 480px) {
       padding-right:0;

    }
",
                ],
                "defaultProps": undefined,
                "displayName": "Title",
                "render": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Connect(CountControl)
              product={
                Object {
                  "img": "/test.png",
                  "price": 100,
                  "title": "test",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "product": Object {
                "img": "/test.png",
                "price": 100,
                "title": "test",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "
    @media only screen and (max-width: 480px) {
        text-align:center;  
    }
",
            ],
            "defaultProps": undefined,
            "displayName": "Styled(div)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              100,
              " р",
            ],
          },
          "ref": null,
          "rendered": Array [
            100,
            " р",
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "
    font-size:25px;
    margin-left:auto;
    @media only screen and (max-width: 480px) {
        text-align:center;  
    }
",
            ],
            "defaultProps": undefined,
            "displayName": "TotalPrice",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <button
              onClick={[Function]}
            >
              Удалить
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Удалить",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Удалить",
            "type": "button",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "
    font-size:25px;
    margin-left:auto;
    @media only screen and (max-width: 480px) {
        text-align:center;  
    }
",
            ],
            "defaultProps": undefined,
            "displayName": "Styled(div)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": "div",
        "__emotion_forwardProp": undefined,
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          "
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    align-items: center;
    margin-bottom:30px;
    @media only screen and (max-width: 480px) {
        display:block;   
    }
",
        ],
        "defaultProps": undefined,
        "displayName": "Styled(div)",
        "render": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
